docs/agent.rst

```diff
@@ -3,34 +3,85 @@
 Micro XRCE-DDS Agent
 ====================
 
-*Micro XRCE-DDS Agent* acts as a server between the DDS Network and *Micro XRCE-DDS Clients*.
-Agents receive messages containing operations from clients.
-Also agents keep track of the clients and the entities they create.
-The Agent uses the entities to interact with the DDS Global Data Space on behalf of the client.
+*Micro XRCE-DDS Agent* acts as a server between the DDS Network and *Micro XRCE-DDS Clients* applications.
+*Agents* receive messages containing operations from *Clients*.
+Also *Agents* keep track of the *Clients* and the entities they create.
+The *Agent* uses the entities to interact with the DDS Global Data Space on behalf of the *Clients*.
+
+The communication between a *Client* and an *Agent* currently supports UDP, TCP and Serial (dependent on the platform).
+While it is running, the *Agent* will attend any received request from the *Clients* and answers back with the result of that request.
+
+Configuration
+-------------
+
+There are several configuration parameters which can be set at **compile time** in order to configure the *Micro RTPS Agent*.
+These parameters can be selected as CMake flags (``-D<parameter>=<value>``) before the compilation.
+The following is a list of the aforementioned parameters:
+
+``CONFIG_RELIABLE_STREAM_DEPTH``
+    Specify the history of the reliable streams (default 16).
+
+``CONFIG_BEST_EFFORT_STREAM_DEPTH``
+    Specify the history of the best-effort streams (default 16).
+
+``CONFIG_HEARTBEAT_PERIOD``
+    Specify the ``HEARTBEAT`` message period in millisecond (default 200).
+
+``CONFIG_SERIAL_TRANSPORT_MTU``
+    Specify the `Maximum Transmission Unit` able to send and receive by Serial (default 512).
+
+``CONFIG_UDP_TRANSPORT_MTU``
+    Specify the `Maximum Transmission Unit` able to send and receive by UDP (default 512).
+
+``CONFIG_TCP_TRANSPORT_MTU``
+    Specify the `Maximum Transmission Unit` able to send and receive by TCP (default 512).
+
+``CONFIG_TCP_MAX_CONNECTIONS``
+    Specify the maximum number of connections which the *Agent* is able to manage (default 100).
+
+``CONFIG_TCP_MAX_BACKLOG_CONNECTIONS``
+    Specify the maximum number of incoming connections which the *Agent* is able to manage (default 100).
```

Comment:
What's exactly the difference with the previous option?

Reply:
The first one refers to connection established with different Client, and the second one (incoming) connection pending to establish. I change it a bit.